lib/adapters/autocomplete-adapter.ts

```diff
@@ -133,7 +133,17 @@ export default class AutocompleteAdapter {
     }
 
     const filtered = !(request.prefix === "" || (triggerChar !== "" && triggerOnly))
-    return filtered ? filter(suggestions, request.prefix, { key: "filterText" }) : suggestions
+    if (filtered) {
+      // filter the suggestions who have `filterText` property
+      const validSuggestions = suggestions.filter((sgs) => typeof sgs.filterText === "string") as Suggestion[] &
```

Comment:
Probably, I can move this check to Zadeh itself.

Reply:
Ya I would think zadeh should treat objects that don't have the `key` the same as an object having an empty string as the key.

Reply:
But now that I think this shouldn't be done. For example, `StringArrayFilterer` expects an array of strings, and no check is done to ensure that every element is a string. So, the same thing should be valid for `ObjectArrayFilterer` for consistency and also performance. I think it is the responsibility of the user to conform to the type-safety of Zadeh, not the reverse.